sdk/core/azure-core/inc/azure/core/base64.hpp

```diff
@@ -38,6 +39,21 @@ namespace Azure { namespace Core {
      */
     static std::string Base64Encode(const std::vector<uint8_t>& data);
 
+    /**
+     * @brief Encodes the vector of binary data into UTF-8 encoded text represented as Base64.
+     *
+     * @param data The binary data to be encoded.
+     * @param length The length of the binaryData parameter.
+     * @return The UTF-8 encoded text in Base64.
+     */
+    static std::string Base64Encode(uint8_t const* const data, size_t length);
+
+    template <size_t length>
+    static std::string Base64Encode(std::array<uint8_t, length> const& data)
```

Comment:
The current public method `Base64Decode` returns `std::vector<uint8_t>`.  It would make sense to use that as the input for Encoding

```suggestion
    static std::string Base64Encode(std::vector<uint8_t, length> const& data)
```

Reply:
Add doxygen for this as well